VedalAI/neuro-game-sdk
# Add nuget package for Unity mod support

Pack using [`NuGet/pack.ps1`](https://github.com/VedalAI/neuro-game-sdk/blob/f6d37564373c257b39a79fbffe0358e88e47a756/Unity/NuGet/pack.ps1) (CI-friendly) then upload to a NuGet feed of your choice through whatever means convenient to you (package is written to `NuGet/bin/Release/VedalAI.NeuroSdk.Unity.w.x.y.z.nupkg`, where `w.x.y.z` is the version).
The package will detect if the user doesn't have a Unity `PackageReference` and will stop the build, hiding the 999 errors that would otherwise show up in the injected `.cs` files.

### How to test
- Pack the package at least once
- Take the `VedalAI.NeuroSdk.Unity.w.x.y.z.nupkg` and put it in `%NUGET_PACKAGES%\vedalai.neurosdk.unity\w.x.y.z\`
  - (if you don't have the `NUGET_PACKAGES` env var, your nuget packages should be in `%USER_PROFILE%\.nuget\packages`)
- Open the `.nupkg` as an archive (with WinRar or 7zip or what have you)
- Extract it to the `w.x.y.z` directory; it should look like this (there may be other files; they aren't necessary):

![screenshot](https://github.com/user-attachments/assets/7b10e854-0f23-4cc1-9227-e74e46ce876e)
- Create a new file called `.nupkg.metadata` with the following contents (or take a copy from any other package):
```json
{
  "version": 1,
  "contentHash": "",
}
```

You may need to reload the project and/or restart your IDE for the local package to be detected.

Then, try to build [`NuGet/Example/Example.csproj`](https://github.com/VedalAI/neuro-game-sdk/blob/f6d37564373c257b39a79fbffe0358e88e47a756/Unity/NuGet/Example/Example.csproj), play around with removing the `PackageReference` for `UnityEngine.Modules`, etcetera.
